internal/db/disk/submission.go

```diff
@@ -69,8 +70,27 @@ func (this *backend) SaveSubmissions(course *model.Course, submissions []*model.
 	return this.saveSubmissionsLock(course, submissions, true)
 }
 
+// Compute the next submission ID based on the current time.
+// The next submission ID will be the largest ID for the user.
 func (this *backend) GetNextSubmissionID(assignment *model.Assignment, email string) (string, error) {
 	submissionID := time.Now().Unix()
+	previousStringID, err := this.getMostRecentSubmissionID(assignment, email)
+	if err != nil {
+		return "", err
+	}
+
+	if previousStringID != "" {
+		previousSubmissionID, err := strconv.Atoi(previousStringID)
+		if err != nil {
+			return "", err
+		}
+
+		// Ensure the next submission ID is the largest.
+		if int64(previousSubmissionID) >= submissionID {
+			submissionID = int64(previousSubmissionID) + 1
+		}
+	}
```

Comment:
Doesn't the user context lock during grading and a monotomic clock ensure that the current ID is at least tied with the largest?
https://github.com/edulinq/autograder-server/blob/main/internal/grader/grade.go#L64

Is this necessary?